modules/amq-launch/added/launch.sh

```diff
@@ -57,57 +59,74 @@ function configureSSL() {
-          fi
-          ;;
-        "mqtt")
-acceptors="${acceptors}            <acceptor name=\"mqtt\">tcp://0.0.0.0:1883?tcpSendBufferSize=1048576;tcpReceiveBufferSize=1048576;protocols=MQTT;useEpoll=true;connectionsAllowed=${connectionsAllowed}</acceptor>\n"
-          if sslEnabled ; then
-acceptors="${acceptors}            <acceptor name=\"mqtt\">tcp://0.0.0.0:8883?tcpSendBufferSize=1048576;tcpReceiveBufferSize=1048576;protocols=MQTT;useEpoll=true;connectionsAllowed=${connectionsAllowed};sslEnabled=true;keyStorePath=${keyStorePath};keyStorePassword=${keyStorePassword}</acceptor>\n"
-          fi
-          ;;
-        "amqp")
-acceptors="${acceptors}            <acceptor name=\"amqp\">tcp://0.0.0.0:5672?tcpSendBufferSize=1048576;tcpReceiveBufferSize=1048576;protocols=AMQP;useEpoll=true;amqpCredits=1000;amqpMinCredits=300;connectionsAllowed=${connectionsAllowed}</acceptor>\n"      
-          if sslEnabled ; then
-    acceptors="${acceptors}            <acceptor name=\"amqp\">tcp://0.0.0.0:5671?tcpSendBufferSize=1048576;tcpReceiveBufferSize=1048576;protocols=AMQP;useEpoll=true;amqpCredits=1000;amqpMinCredits=300;connectionsAllowed=${connectionsAllowed};sslEnabled=true;keyStorePath=${keyStorePath};keyStorePassword=${keyStorePassword}</acceptor>\n"
-          fi
-          ;;
-        "stomp")
-acceptors="${acceptors}            <acceptor name=\"stomp\">tcp://0.0.0.0:61613?tcpSendBufferSize=1048576;tcpReceiveBufferSize=1048576;protocols=STOMP;useEpoll=true;connectionsAllowed=${connectionsAllowed}</acceptor>\n"
-          if sslEnabled ; then
-    acceptors="${acceptors}            <acceptor name=\"stomp\">tcp://0.0.0.0:61612?tcpSendBufferSize=1048576;tcpReceiveBufferSize=1048576;protocols=STOMP;useEpoll=true;connectionsAllowed=${connectionsAllowed};sslEnabled=true;keyStorePath=${keyStorePath};keyStorePassword=${keyStorePassword}</acceptor>\n"
-          fi
-          ;;
-        "hornetq")
-acceptors="${acceptors}            <acceptor name=\"hornetq\">tcp://0.0.0.0:5445?protocols=HORNETQ,STOMP;useEpoll=true;connectionsAllowed=${connectionsAllowed}</acceptor>\n"
-          ;;
-      esac
-    done
-    sed -i -ne "/<acceptors>/ {p; i $acceptors" -e ":a; n; /<\/acceptors>/ {p; b}; ba}; p" ${instanceDir}/etc/broker.xml
+    	IFS=',' read -a protocols <<< $(find_env "AMQ_TRANSPORTS" "openwire,amqp,stomp,mqtt,hornetq")
+    	connectionsAllowed=$(find_env "AMQ_MAX_CONNECTIONS" "1000")
+
+    	if [ "${#protocols[@]}" -ne "0" ]; then
+    		acceptors=""
+    		for protocol in ${protocols[@]}; do
+    		case "${protocol}" in
+        	"openwire")
+				acceptors="${acceptors}            <acceptor name=\"artemis-ssl\">tcp://${ACCEPTOR_IP}:61617?tcpSendBufferSize=1048576;tcpReceiveBufferSize=1048576;protocols=CORE,AMQP,STOMP,HORNETQ,MQTT,OPENWIRE;useEpoll=true;amqpCredits=1000;amqpLowCredits=300;connectionsAllowed=${connectionsAllowed};sslEnabled=true;keyStorePath=${keyStorePath};keyStorePassword=${keyStorePassword}</acceptor>\n"
```

Comment:
@vbusch does this change completely intentionally omit non-secured variant of `acceptors`?

Reply:
The launch scripts calls '/opt/amq/bin/artemis create', which populates the non-secure acceptors.  Afterwards, updateAcceptors adds the secure ones.